src/pypi_attestations/_impl.py

```diff
@@ -331,3 +331,85 @@ def _ultranormalize_dist_filename(dist: str) -> str:
         return f"{name}-{ver}.tar.gz"
     else:
         raise ValueError(f"unknown distribution format: {dist}")
+
+
+class Publisher(BaseModel):
+    """Publisher as defined in PEP 740."""
+
+    kind: str
+    """
+    The kind of Trusted Publisher.
+    """
+
+    claims: Union[object, None] = Field(...)  # noqa: UP007
```

Comment:
This is a bit annoying here. 

`pydantic` reports an error if we use the `x|y` notation but `ruff` reports another one if we use the `Union` notation because we use the `from future import __annotations__`.

The other option here is to depend on `eval_type_backport` or to drop support for Python3.9.

Reply:
looking at the docs, this is a [known problem](https://docs.astral.sh/ruff/rules/non-pep604-annotation/#why-is-this-bad) when using runtime annotations (Pydantic) and `__future__.annotations`, and we can fix it by enabling [this option](https://docs.astral.sh/ruff/settings/#lint_pyupgrade_keep-runtime-typing) in `pyproject.toml`:

```toml
[tool.ruff.lint.pyupgrade]
# Preserve types, even if a file imports `from __future__ import annotations`.
keep-runtime-typing = true
```

Reply:
> The other option here is to depend on eval_type_backport or to drop support for Python3.9.

Hmm -- we could probably get away with this, since Warehouse is on 3.11 (IIRC) and `gh-action-pypi-publish` is on 3.12.

Reply:
I've put this change in another PR to keep this one focused: https://github.com/trailofbits/pypi-attestations/pull/37